File: Problems/0605. (E) Can Place Flowers/Java/solution.java
class Solution {
  public boolean canPlaceFlowers(int[] flowerbed, int n) {
    // brute force (greedy)

    // sanity check
    if (n == 0)
      return true;

    int[] fb = Arrays.copyOf(flowerbed, flowerbed.length);
    int planted = 0;

    for (int i = 0; i < fb.length; i++) {
      if (fb[i] == 0) {
        // check left and right plots
        boolean l = i == 0 || fb[i-1] == 0 ? true : false;
        boolean r = i == fb.length-1 || fb[i+1] == 0 ? true : false;
        // plant if plot available
        if (l && r) {
          fb[i] = 1;
          planted++;
          if (planted >= n)
            return true;
        }
      }
    }
    return planted >= n;
  }
}

File: Problems/0605. (E) Can Place Flowers/Python3/solution.py
class Solution:
  def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
    # brute force (greedy)

    # sanity check
    if not n:
      return True

    fb = copy.deepcopy(flowerbed)
    planted = 0

    for i in range(len(fb)):
      if not fb[i]:
        # check left and right plots
        l = True if not i or not fb[i-1] else False
        r = True if i == len(fb) - 1 or not fb[i+1] else False
        # plant if plot available
        if l and r:
          fb[i] = 1
          planted += 1
          if planted >= n:
            return True
      
    return planted >= n

